site/en/articles/renderingng-fragmentation/index.md

```diff
@@ -0,0 +1,204 @@
+---
+layout: 'layouts/blog-post.njk'
+title: 'RenderingNG deep-dive: LayoutNG block fragmentation'
+description: >
+  In 2021, we will largely complete the process of designing, building and shipping RenderingNG, a next-generation rendering architecture for Chromium.
+subhead: >
+  Ready for the next generation of web content
+date: 2022-12-08
+authors:
+  - mortenstenshorne
+tags:
+  - rendering
+---
+{% Aside %}
+This post is a part of a series on the Chromium rendering engine. Check out the rest of the series to learn more about [RenderingNG](/articles/renderingng/), the [RenderingNG architecture](/blog/renderingng-architecture/), [key data structures](/blog/renderingng-data-structures/), [VideoNG](/blog/videong/), [LayoutNG](/blog/layoutng/)
+and [BlinkNG](/blog/blinkng/).
+{% endAside %}
+
+I'm Morten Stenshorne, a layout engineer on the Blink rendering team at Google. I've been involved in browser engine development since the early 2000s, and I've had a lot of fun, such as helping make the [acid2 test](https://en.wikipedia.org/wiki/Acid2#Timeline_of_passing_applications) pass in the Presto engine (Opera 12 and earlier), and reverse-engineering other browsers to fix table layout in Presto. I've also spent more of those years than I'd like to admit on block fragmentation, and, in particular, [multicol](https://www.w3.org/TR/css-multicol-1/) in Presto, WebKit, and Blink. During the past few years at Google I have mainly been focusing on leading the work of adding block fragmentation support to [LayoutNG](/articles/layoutng/). Join me in this deep-dive into the block fragmentation implementation, as it may very well be the last time that I implement block fragmentation. :)
+
+## What is block fragmentation?
+
+[Block fragmentation](https://www.w3.org/TR/css-break-3/) is about splitting a CSS block-level box (such as a section or paragraph) into multiple fragments when it doesn't fit as a whole inside one fragment container (called a [_fragmentainer_](https://www.w3.org/TR/css-break-3/#fragmentainer). A fragmentainer is not an element, but represents a column in multi-column layout, or a page in paged media. For fragmentation to happen, the content needs to be inside a fragmentation context. A fragmentation context is most commonly established by a multi-column container (content will be split into columns), or when printing (content will be split into pages). A long paragraph with many lines may need to be split into multiple fragments, so that the first lines are placed in the first fragment, and the remaining lines are placed in subsequent fragments. 
```

Comment:
```suggestion
[Block fragmentation](https://www.w3.org/TR/css-break-3/) is about splitting a CSS block-level box (such as a section or paragraph) into multiple fragments when it doesn't fit as a whole inside one fragment container (called a [_fragmentainer_](https://www.w3.org/TR/css-break-3/#fragmentainer). A fragmentainer is not an element, but represents a column in multi-column layout, or a page in paged media. For fragmentation to happen, the content needs to be inside a *fragmentation context*. A fragmentation context is most commonly established by a multi-column container (content will be split into columns), or when printing (content will be split into pages). A long paragraph with many lines may need to be split into multiple fragments, so that the first lines are placed in the first fragment, and the remaining lines are placed in subsequent fragments. 
```